src/cdk/a11y/focus-monitor/focus-monitor.ts

```diff
@@ -552,6 +567,40 @@ export class FocusMonitor implements OnDestroy {
 
     return results;
   }
+
+  /**
+   * Returns whether an interaction is likely to have come from the user clicking the `label` of
+   * an `input` or `textarea` in order to focus it.
+   * @param focusEventTarget Target currently receiving focus.
+   */
```

Comment:
A much simpler alternative I was considering was to listen to `click` events inside the `InputModalityDetector` instead and set the modality to `mouse` again. I decided against it, because it would've meant that `modalityDetected` would emit two consecutive `mouse` interactions on any click which can be annoying for consumers. Also this seems very much like a focus issue so it makes more sense to handle it in the `FocusMonitor` than the `InputModalityDetector`.